src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNITcpHandle.cs

```diff
@@ -143,6 +146,12 @@ public SNITCPHandle(string serverName, int port, long timerExpire, object callba
                 }
 
                 _socket = connectTask.Result;
+                if (_socket == null)
```

Comment:
Correction, looks like ReceiveTimeout only applies to Receive() calls. Looks like the only "timeout" that's applicable to socket connections is just a Wait like this one. So in case of timeout here the sockets will be closed upon Dispose during garbage collection.